src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/AmendApplicationConsentedMidHandlerTest.java

```diff
@@ -1,71 +1,256 @@
+import uk.gov.hmcts.reform.finrem.caseorchestration.controllers.GenericAboutToStartOrSubmitCallbackResponse;
 import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.Address;
 import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.YesOrNo;
 import uk.gov.hmcts.reform.finrem.caseorchestration.service.ConsentOrderService;
 import uk.gov.hmcts.reform.finrem.caseorchestration.service.InternationalPostalService;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.stream.Stream;
 
-import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.TestConstants.AUTH_TOKEN;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType.ABOUT_TO_START;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType.MID_EVENT;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType.AMEND_APP_DETAILS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType.SOLICITOR_CREATE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType.UPDATE_CONTACT_DETAILS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType.CONSENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType.CONTESTED;
 
 @ExtendWith(MockitoExtension.class)
 class AmendApplicationConsentedMidHandlerTest {
 
     @InjectMocks
     private AmendApplicationConsentedMidHandler handler;
-    static final String AUTH_TOKEN = "4d73f8d4-2a8d-48e2-af91-11cbaa642345";
-
     @Mock
     private ConsentOrderService consentOrderService;
     @Mock
     private InternationalPostalService postalService;
+    @Mock
+    private ObjectMapper objectMapper;
+
+
+    private static Stream<Arguments> testCanHandle() {
+        return Stream.of(
+            Arguments.of(MID_EVENT, CONSENTED, AMEND_APP_DETAILS, true),
+            Arguments.of(ABOUT_TO_START, CONSENTED, SOLICITOR_CREATE, false),
+            Arguments.of(MID_EVENT, CONTESTED, AMEND_APP_DETAILS, false),
+            Arguments.of(MID_EVENT, CONSENTED, SOLICITOR_CREATE, false),
+            Arguments.of(MID_EVENT, CONSENTED, UPDATE_CONTACT_DETAILS, false)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void testCanHandle(CallbackType callbackType, CaseType caseType, EventType eventType, boolean expected) {
```

Comment:
take a look at `RemoveUserCaseAccessAboutToSubmitHandlerTest`.
you may do this - 
        assertCanHandle(handler,
            Arguments.of(CallbackType.ABOUT_TO_SUBMIT, CaseType.CONSENTED, EventType.REMOVE_USER_CASE_ACCESS),
            Arguments.of(CallbackType.ABOUT_TO_SUBMIT, CaseType.CONTESTED, EventType.REMOVE_USER_CASE_ACCESS)
        );